onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2166,10 +2177,14 @@ def aten_feature_dropout(input: TensorType, p: float, train: bool) -> TensorType
     raise NotImplementedError()
 
 
-def aten_fill(self: TensorType, value: TensorType) -> TensorType:
+@torch_op("aten::fill")
+def aten_fill(self: TTensor, value: TTensor) -> TTensor:
     # fill.Tensor(Tensor self, Tensor value) -> Tensor
-
-    raise NotImplementedError()
+    # after fill, the self Tensor should keep origianl type
+    shape = op.Shape(self)
+    result = op.Expand(value, shape)
+    result = op.CastLike(result, self)
```

Comment:
```suggestion
    expanded = op.Expand(value, shape)
    result = op.CastLike(expanded, self)
```

so we avoid reusing names which can be confusing in the graph